operators/volume_loader/dual_type_receiver.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef VOLUME_LOADER_DUAL_TYPE_RECEIVER
+#define VOLUME_LOADER_DUAL_TYPE_RECEIVER
+
+#include <pybind11/stl.h>
+#include "gil_guarded_pyobject.hpp"
+
+namespace holoscan::ops {
+
+template <typename T>
+struct is_shared_ptr : std::false_type {};
+template <typename T>
+struct is_shared_ptr<std::shared_ptr<T>> : std::true_type {};
+
+/*
+ * Any object emitted from a Python operator on a port (except Tensor) is tracked by the
+ * GILGuardedObject. With that we can detect if the object received by the C++ operator
+ * is coming from Python and do the conversion to a C++ type.
+ * This needs to be moved to Holoscan SDK.
+ * See discussion https://nvidia.slack.com/archives/C03E643SK89/p1701875573021209.
+ */
```

Comment:
This is actually a hack I implemented. I don't think this should be added here. @grlee77 implemented support for this in Holoscan SDK 2.1. We should require .2.1 for the app and remove this and the `gil_guarded_object.hpp` file.

Reply:
Note that we do currently require the `volume_rendering` application is backwards compatible with Holoscan SDK v1.0.3 for NVIDIA AI Enterprise related reasons. Maybe there is a way to default to the 2.1 functionality with include guards for backwards compatibility? Happy to chat further internally.

Reply:
Thank you @AndreasHeumann! I've removed `dual_type_receiver` and `gil_guarded_pyobject`.